src/System/System.php

```diff
@@ -118,100 +109,98 @@ public static function getHostname(): string
 
     /**
      * Checks if the system is running on an ARM architecture.
-     *
+     * 
      * @return bool
      */
     public static function isArm(): bool
     {
-        return (bool) preg_match(self::RegExARM, self::getArch());
+        return !!preg_match(self::RegExARM, self::getArch());
     }
 
     /**
      * Checks if the system is running on an X86 architecture.
-     *
+     * 
      * @return bool
      */
     public static function isX86(): bool
     {
-        return (bool) preg_match(self::RegExX86, self::getArch());
+        return !!preg_match(self::RegExX86, self::getArch());
     }
 
     /**
      * Checks if the system is running on an PowerPC architecture.
-     *
+     * 
      * @return bool
      */
     public static function isPPC(): bool
     {
-        return (bool) preg_match(self::RegExPPC, self::getArch());
+        return !!preg_match(self::RegExPPC, self::getArch());
     }
 
     /**
-     * Checks if the system is the passed architecture.
+     * Checks if the system is the passed architecture. 
      * You should pass `System::X86`, `System::PPC`, `System::ARM` or an equivalent string.
-     *
-     * @param  string  $arch
+     * 
+     * @param string $arch
+     * 
      * @return bool
-     *
+     * 
      * @throws Exception
      */
     public static function isArch(string $arch): bool
     {
         switch ($arch) {
-            case self::X86:
-                return self::isX86();
-                break;
-            case self::PPC:
-                return self::isPPC();
-                break;
-            case self::ARM:
-                return self::isArm();
-                break;
-
-            default:
-                throw new Exception("'{$arch}' not found.");
-                break;
+        case self::X86:
+            return self::isX86();
+        case self::PPC:
+            return self::isPPC();
+        case self::ARM:
+            return self::isArm();
+
+        default:
+            throw new Exception("'{$arch}' not found.");
         }
     }
 
     /**
      * Gets the system's total amount of CPU cores.
-     *
-     * @return int
-     *
+     * 
+     * @return string|int
+     * 
      * @throws Exception
      */
-    public static function getCPUCores(): int
+    public static function getCPUCores(): string|int
     {
         switch (self::getOS()) {
             case 'Linux':
                 $cpuinfo = file_get_contents('/proc/cpuinfo');
-                preg_match_all('/^processor/m', $cpuinfo, $matches);
-
-                return count($matches[0]);
+                if(is_string($cpuinfo)==1)
+                {preg_match_all('/^processor/m', $cpuinfo, $matches);
```

Comment:
This doesn't match our code style. Please use `composer format` to ensure it matches our style.